client/platform/desktop/frontend/store/settings.ts

```diff
@@ -30,30 +30,38 @@ function isSettings(s: any): s is Settings {
   return true;
 }
 
+// Settings initialization involves a handoff between renderer and background.
+// This function should be called at application startup.
 async function init() {
-  const settingsStr = window.localStorage.getItem(SettingsKey);
-  let settingsvalue = await getDefaultSettings();
+  // Client asks for default settings and external force overrides from background
+  let settingsValue = await getDefaultSettings();
   try {
-    if (settingsStr) {
-      const maybeSettings = JSON.parse(settingsStr);
-      if (isSettings(maybeSettings)) {
-        settingsvalue = {
-          // Populate from defaults to include any missing properties
-          ...settingsvalue,
-          // Overwrite with explicitly persisted settings
-          ...maybeSettings,
-        };
-      }
+    // Client applies user-configured settings from localstorage
+    const settingsStr = window.localStorage.getItem(SettingsKey) || '{}';
+    const maybeSettings = JSON.parse(settingsStr);
+    if (isSettings(maybeSettings)) {
+      settingsValue = {
+        // Populate from defaults to include any missing properties
+        ...settingsValue,
+        // Overwrite with explicitly persisted settings
+        ...maybeSettings,
```

Comment:
This could be a two-part because the correct `localSettings` aren't written unless you save the data So see the other comment further down.
If you don't have a DIVE_VIAME_INSTALL_PATH set and have a previous version when you load up the data it will set override to `{}` in the local settings.
If you then set DIVE_VIAME_INSTALL_PATH afterwards and load it up, the locaStorage will always have the `overrides: {}`.  I'm guessing because your spreading the `...maybeSettings` over the top of `...settingsValue` so it is wiping out the default `overrides:` with the blank one.
Even if you get the correct `settings` set here it doesn't update the localSettings for the client.  I don't know if that is required during this part.

Reply:
I was a bit confused what you meant at first, but after understanding this is a great catch. Really appreciate you pointing out this major issue. I was actually having trouble with getting the environment variables to load in and thought I was going insane, but this was actually the issue.

A quick temporary fix I'm trying for now is just to delete the `overrides` property from `maybeSettings` (since it's a JSON and not a proper type). Then spreading it on top of `settingsValue` will overwrite everything except the overrides. I'm open to any better solutions.

Reply:
I think you might want to check the `overrides` to ensure the `Object.keys(maybeSettings).length !== 0` or something similar before deleting it or setting it to undefined.  If there exists the path/readonly settings you probably want them to spread and overwrite the defaults.

Reply:
Could we add a little comment above the `delete` line about what that's for?